src/Framework/TestBuilder.php

```diff
@@ -56,151 +56,157 @@ public function build(\ReflectionClass $theClass, string $name): Test
 
         // TestCase() or TestCase($name)
         if (\count($parameters) < 2) {
-            $test = new $className;
+            $test = $this->buildTestWithoutData($className);
         } // TestCase($name, $data)
         else {
             try {
                 $data = TestUtil::getProvidedData(
                     $className,
-                    $name
+                    $methodName
                 );
             } catch (IncompleteTestError $e) {
                 $message = \sprintf(
                     'Test for %s::%s marked incomplete by data provider',
                     $className,
-                    $name
+                    $methodName
                 );
-
-                $_message = $e->getMessage();
-
-                if (!empty($_message)) {
-                    $message .= "\n" . $_message;
-                }
-
-                $data = new IncompleteTestCase($className, $name, $message);
+                $message = $this->appendExceptionMessageIfAvailable($e, $message);
+                $data    = new IncompleteTestCase($className, $methodName, $message);
             } catch (SkippedTestError $e) {
                 $message = \sprintf(
                     'Test for %s::%s skipped by data provider',
                     $className,
-                    $name
+                    $methodName
                 );
-
-                $_message = $e->getMessage();
-
-                if (!empty($_message)) {
-                    $message .= "\n" . $_message;
-                }
-
-                $data = new SkippedTestCase($className, $name, $message);
+                $message = $this->appendExceptionMessageIfAvailable($e, $message);
+                $data    = new SkippedTestCase($className, $methodName, $message);
             } catch (\Throwable $t) {
                 $message = \sprintf(
                     'The data provider specified for %s::%s is invalid.',
                     $className,
-                    $name
+                    $methodName
                 );
-
-                $_message = $t->getMessage();
-
-                if (!empty($_message)) {
-                    $message .= "\n" . $_message;
-                }
-
-                $data = new WarningTestCase($message);
+                $message = $this->appendExceptionMessageIfAvailable($t, $message);
+                $data    = new WarningTestCase($message);
             }
 
             // Test method with @dataProvider.
             if (isset($data)) {
-                $test = new DataProviderTestSuite(
-                    $className . '::' . $name
+                $test = $this->buildDataProviderTestSuite(
+                    $methodName,
+                    $className,
+                    $data,
+                    $runTestInSeparateProcess,
+                    $preserveGlobalState,
+                    $runClassInSeparateProcess,
+                    $backupSettings
                 );
-
-                if (empty($data)) {
```

Comment:
I removed this if as `TestUtil::getProvidedData` never returns empty array this code is unreachable. To make sure it breaks if someone removes the empty array check I implemented the test `TestBuilderTest.php::testCreateWithEmptyData`